lms/djangoapps/course_blocks/transformers/library_content.py

```diff
@@ -177,3 +180,66 @@ def publish_event(event_name, result, **kwargs):
             format_block_keys,
             publish_event,
         )
+
+
+class ContentLibraryOrderTransformer(BlockStructureTransformer):
+    """
+    A transformer that manipulates the block structure by modifying the order of the
+    selected blocks within a library_content module to match the order of the selections
+    made by the ContentLibraryTransformer or the corresponding XBlock. So this transformer
+    requires the selections for the randomized content block to be already
+    made either by the ContentLibraryTransformer or the XBlock.
+
+    Staff users are *not* exempted from library content pathways.
+    """
+    WRITE_VERSION = 1
+    READ_VERSION = 1
+
+    @classmethod
+    def name(cls):
+        """
+        Unique identifier for the transformer's class;
+        same identifier used in setup.py
+        """
+        return "library_content_randomize"
+
+    @classmethod
+    def collect(cls, block_structure):
+        """
+        Collects any information that's necessary to execute this
+        transformer's transform method.
+        """
+        # There is nothing to collect
+        pass  # pylint:disable=unnecessary-pass
+
+    def transform(self, usage_info, block_structure):
+        """
+        Transforms the order of the children of the randomized content block
+        to match the order of the selections made and stored in the XBlock 'selected' field.
+        """
+        for block_key in block_structure:
+            if block_key.block_type != 'library_content':
+                continue
+
+            library_children = block_structure.get_children(block_key)
+
+            if library_children:
+                state_dict = get_student_module_as_dict(usage_info.user, usage_info.course_key, block_key)
+                current_children_blocks = set(block.block_id for block in library_children)
+                current_selected_blocks = set(item[1] for item in state_dict['selected'])
+
+                # As the selections should have already been made by the ContentLibraryTransformer,
+                # the current children of the library_content block should be the same as the stored
+                # selections. If they aren't, some other transformer that ran before this transformer
+                # has modified those blocks (for example, content gating may have affected this). So do not
+                # transform the order in that case.
+                if current_children_blocks != current_selected_blocks:
+                    logger.info(
+                        u'Mismatch between the children of %s in the stored state and the actual children for user %s. '
+                        'Continuing without order transformation.',
+                        str(block_key),
+                        usage_info.user.username
+                    )
```

Comment:
Hey @lgp171188. This statement is blowing up our logs -- we're seeing 250k logs/hour from it.

Is it alerting us of something important that we should fix? Or can we remove it or turn it to `debug` level, which will exclude it from our production logs?

Reply:
@kdmccormick, this PR was merged after 3 reverts due to issues in this transformer and that is why this logging statement was added. But now that the PR is merged and the changes have worked okay in production for a while now, it should be okay to remove it or turn it to the `debug` level. 👍🏼 Sorry for causing this issue!

Reply:
No sweat! Can you give https://github.com/edx/edx-platform/pull/25065 a look? It sets the logging level to debug.